backend/databases/postgres.py

```diff
@@ -0,0 +1,64 @@
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker, Session
+
+from utils.logger import LoggerFactory
+from settings import Constants, Secrets
+
+logger = LoggerFactory().get_logger(__name__)
+
+
+class PostgresConnector:
+    """
+    Pattern: Singleton
+    Purpose: Create a single instance of the database connection
+    """
+
+    _instance = None
+
+    def __init__(self):
+        if PostgresConnector._instance is None:
+            PostgresConnector._instance = self.__create_engine()
+
+    @classmethod
+    def get_instance(cls):
+        """
+        Get the instance of the database connection
+        """
+        if cls._instance is None:
+            cls._instance = cls().__create_engine()
+        return cls._instance
+
+    @staticmethod
+    def __create_engine():
+        """
+        Create the database connection if there is no any existing connection
+        """
+        try:
+            return create_engine(
+                Secrets.POSTGRES_URI,
```

Comment:
How did you validate the `Secrets.POSTGRES_URI`?

Reply:
What do you mean by validating the `Secrets.POSTGRES_URI` ? The `Secrets.POSTGRES_URI` is a format string where we parse the database credentials such as username, password, etc.